scripts/tunnel-creation.mjs

```
@@ -0,0 +1,351 @@
+    });
+  }
+
+  /**
+   * Create tunnel for a single device
+   * @param {Device} device - Device object
+   * @param {import('tls').ConnectionOptions} tlsOptions - TLS options
+   * @returns {Promise<import('appium-ios-remotexpc').TunnelResult & { socket?: any; socketInfo?: import('appium-ios-remotexpc').SocketInfo }>} Tunnel result
+   */
+  async createTunnelForDevice(device, tlsOptions) {
+    const udid = device.Properties.SerialNumber;
+
+    try {
+      log.info(`\n--- Processing device: ${udid} ---`);
+      log.info(`Device ID: ${device.DeviceID}`);
+      log.info(`Connection Type: ${device.Properties.ConnectionType}`);
+      log.info(`Product ID: ${device.Properties.ProductID}`);
+
+      log.info('Creating lockdown service...');
+      const {lockdownService, device: lockdownDevice} = await createLockdownServiceByUDID(udid);
+      log.info(`Lockdown service created for device: ${lockdownDevice.Properties.SerialNumber}`);
+
+      log.info('Starting CoreDeviceProxy...');
+      const {socket} = await startCoreDeviceProxy(
+        lockdownService,
+        lockdownDevice.DeviceID,
+        lockdownDevice.Properties.SerialNumber,
+        tlsOptions,
+      );
+      log.info('CoreDeviceProxy started successfully');
+
+      log.info('Creating tunnel...');
+      const tunnel = await TunnelManager.getTunnel(socket);
+      log.info(`Tunnel created for address: ${tunnel.Address} with RsdPort: ${tunnel.RsdPort}`);
+
+      let packetStreamPort;
+      try {
+        packetStreamPort = this.packetStreamBasePort++;
+        const packetStreamServer = new PacketStreamServer(packetStreamPort);
+        await packetStreamServer.start();
+
+        const consumer = packetStreamServer.getPacketConsumer();
+        if (consumer) {
+          tunnel.addPacketConsumer(consumer);
+        }
+
+        this.packetStreamServers.set(udid, packetStreamServer);
+
+        log.info(`Packet stream server started on port ${packetStreamPort}`);
+      } catch (err) {
```

let it throw an exception. This wrapper masks the actual stacktrace, which makes the code harder to debug